test/boost/big_decimal_test.cc

```diff
@@ -203,3 +203,112 @@ BOOST_AUTO_TEST_CASE(test_big_decimal_sub) {
     test_sub("9999999999999999999999999999999999999", "-1.000e0", "10000000000000000000000000000000000000.000");
     test_sub("+10.", "1.e+1", "0");
 }
+
+
+BOOST_AUTO_TEST_CASE(test_boost_multiprecision_sign) {
+    namespace bmp = boost::multiprecision;
+
+    // Validate assumption made about boost::multiprecision::sign
+    BOOST_REQUIRE_EQUAL(bmp::sign(bmp::cpp_int(-10000)), -1);
+    BOOST_REQUIRE_EQUAL(bmp::sign(bmp::cpp_int(-55545456)), -1);
+    BOOST_REQUIRE_EQUAL(bmp::sign(bmp::cpp_int(-1)), -1);
+    BOOST_REQUIRE_EQUAL(bmp::sign(bmp::cpp_int(0)), 0);
+    BOOST_REQUIRE_EQUAL(bmp::sign(bmp::cpp_int(1)), 1);
+    BOOST_REQUIRE_EQUAL(bmp::sign(bmp::cpp_int(10000)), 1);
+    BOOST_REQUIRE_EQUAL(bmp::sign(bmp::cpp_int(55545456)), 1);
+}
+
+BOOST_AUTO_TEST_CASE(test_big_decimal_cmp) {
+    const auto fmt_ordering = [] (std::strong_ordering r) {
+        if (r == std::strong_ordering::less) {
+            return "<";
+        } else if (r == std::strong_ordering::equal) {
+            return "=";
+        } else if (r == std::strong_ordering::greater) {
+            return ">";
+        } else {
+            return "?";
+        }
+    };
+    // Some of the numbers in this test are truly humongous, converting them to string will stress the machine and will make the test output unusable.
+    // So avoid rendering the numbers using big_decimal::to_string(), just print the unscaled value and the scale instead.
```

Comment:
This observation really points to a bug in bid_decimal::to_string() - there is no reason why `1789679678670986897698780986986E99999999` can't be printed like that, in scientific notation, and needs to be written as 100 million digits. This bug in to_string() probably causes https://github.com/scylladb/scylladb/issues/8002 and the fact we use it probably also explains https://github.com/scylladb/scylladb/issues/7972. But you don't need to solve it now.